packages/core/src/types.js

```diff
@@ -121,15 +121,17 @@ export type FormComponentProps = {
   parentContext?: FormContextData,
   showValidationBeforeTouched?: boolean,
   conditionalUpdate?: boolean,
-  disabled?: boolean
+  disabled?: boolean,
+  touched?: boolean
```

Comment:
Rather than adding a `touched` prop here, it might be better to just reuse the `showValidationBeforeTouched` prop. When this is set to `false` all the warning messages will be shown immediately even if the field hasn't been given focus.